pkg/controllers/job/job_controller.go

```diff
@@ -127,53 +122,18 @@ func NewJobController(config *rest.Config) *Controller {
 	cc.jobSynced = cc.jobInformer.Informer().HasSynced
 
 	cc.cmdInformer = vkinfoext.NewSharedInformerFactory(cc.vkClients, 0).Bus().V1alpha1().Commands()
-	cc.cmdInformer.Informer().AddEventHandler(cache.FilteringResourceEventHandler{
-		FilterFunc: func(obj interface{}) bool {
-			switch t := obj.(type) {
-			case *v1corev1.Command:
-				return helpers.ControlledBy(t, helpers.JobKind)
-			case cache.DeletedFinalStateUnknown:
-				if cmd, ok := t.Obj.(*v1corev1.Command); ok {
-					return helpers.ControlledBy(cmd, helpers.JobKind)
-				}
-				runtime.HandleError(fmt.Errorf("unable to convert object %T to *v1.Command", obj))
-				return false
-			default:
-				runtime.HandleError(fmt.Errorf("unable to handle object %T", obj))
-				return false
-			}
-		},
-		Handler: cache.ResourceEventHandlerFuncs{
-			AddFunc: cc.addCommand,
-		},
+	cc.cmdInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: cc.addCommand,
 	})
 	cc.cmdLister = cc.cmdInformer.Lister()
 	cc.cmdSynced = cc.cmdInformer.Informer().HasSynced
 
 	cc.sharedInformers = informers.NewSharedInformerFactory(cc.kubeClients, 0)
 	podInformer := cc.sharedInformers.Core().V1().Pods()
-
-	podInformer.Informer().AddEventHandler(cache.FilteringResourceEventHandler{
-		FilterFunc: func(obj interface{}) bool {
-			switch t := obj.(type) {
-			case *v1.Pod:
-				return helpers.ControlledBy(t, helpers.JobKind)
-			case cache.DeletedFinalStateUnknown:
-				if pod, ok := t.Obj.(*v1.Pod); ok {
-					return helpers.ControlledBy(pod, helpers.JobKind)
-				}
-				runtime.HandleError(fmt.Errorf("unable to convert object %T to *v1.Pod", obj))
-				return false
-			default:
-				runtime.HandleError(fmt.Errorf("unable to handle object %T", obj))
-				return false
-			}
-		},
-		Handler: cache.ResourceEventHandlerFuncs{
-			AddFunc:    cc.addPod,
-			UpdateFunc: cc.updatePod,
-			DeleteFunc: cc.deletePod,
-		},
+	podInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    cc.addPod,
```

Comment:
Ditto

Reply:
This is not needed,  we already have many other checks in event handlers, like this

```
	taskName, found := pod.Annotations[vkbatchv1.TaskSpecKey]
	if !found {
		glog.Infof("Failed to find taskName of Pod <%s/%s>, skipping",
			pod.Namespace, pod.Name)
		return
	}

	jobName, found := pod.Annotations[vkbatchv1.JobNameKey]
	if !found {
		glog.Infof("Failed to find jobName of Pod <%s/%s>, skipping",
			pod.Namespace, pod.Name)
		return
	}

	version, found := pod.Annotations[vkbatchv1.JobVersion]
	if !found {
		glog.Infof("Failed to find jobVersion of Pod <%s/%s>, skipping",
			pod.Namespace, pod.Name)
		return
	}
```

Reply:
Which is faster, just asking.

Reply:
The above check is inevitable, so it's enough, if we check twice, then absolutely double time wasting